src/Toolbox/URL.php

```diff
@@ -74,4 +74,81 @@ final public static function sanitizeURL(?string $url): string
 
         return $url;
     }
+
+    /**
+     * Extract (lowercase) itemtype from a given URL path.
+     *
+     * For example:
+     * - '/front/itemtype.php' will yield 'itemtype'
+     * - '/front/namespace/itemtype.form.php' will yield 'glpi\namespace\itemtype'
+     *
+     * Both .php and .form.php page are supported, and plugins from /plugins or
+     * /marketplace.
+     *
+     * @param string $path The filename of the currently executing script,
+     *                     relative to the document root.
+     *                     For the "http://example.com/foo/bar.php" page, that
+     *                     would be "/foo/bar.php" (= $_SERVER['PHP_SELF']).
+     * @return string|null Null if the itemtype could not be extracted.
+     *
+     * @todo Support custom marketplace and plugins URL.
+     */
+    public static function extractItemtypeFromUrlPath(string $path): ?string
+    {
+        if (self::isPluginUrlPath($path)) {
+            return self::extractPluginItemtypeFromUrlPath($path);
+        } else {
+            return self::extractCoreItemtypeFromUrlPath($path);
+        }
+    }
+
+    private static function isPluginUrlPath(string $path): bool
+    {
+        return
+            str_contains($path, '/front/plugins/')
+            || str_contains($path, '/front/marketplace/')
```

Comment:
It is not correct. URLS are `/plugins/{$plugin_key}/front` or `/marketplace/{$plugin_key}/front`
```suggestion
            str_contains($path, '/front/plugins/')
            || str_contains($path, '/front/marketplace/')
```